networkx/drawing/layout.py

```diff
@@ -1329,8 +1329,10 @@ def forceatlas2_layout(
     if pos is None:
         pos = nx.random_layout(G, dim=dim, seed=seed)
         pos_arr = np.array(list(pos.values()))
+    elif len(pos) == len(G):
+        pos_arr = np.array([pos[node].copy() for node in G])
```

Comment:
This avoids computing `max_pos`, `min_pos`, etc when all node positions have already been specified (the usual case if any have been specified). The case where lengths are the same but nodes differ is covered by `pos[node]` being used in the loop over G when creating pos_arr.